eclair-core/src/test/scala/fr/acinq/eclair/io/PeerSpec.scala

```diff
@@ -379,3 +368,15 @@ class PeerSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike with Paralle
     assert(init.pushAmount === 100.msat)
   }
 }
+
+object PeerSpec {
+
+  def createMockServer(): (ServerSocketChannel, InetSocketAddress) = {
+    val mockServer = ServerSocketChannel.open()
+    // NB: we force 127.0.0.1 (IPv4) because there are issues on ubuntu build machines with IPv6 loopback
+    mockServer.bind(new InetSocketAddress("127.0.0.1", 0))
+    mockServer.configureBlocking(false)
```

Comment:
`configureBlocking(false)` ❤️

Reply:
I discovered that one while I was thinking that the issue was a deadlock on the server. It didn't fix the build, but it's still an improvement :smiley: